src/parser/monk/mistweaver/modules/talents/Lifecycles.tsx

```diff
@@ -61,11 +59,29 @@ class Lifecycles extends Analyzer {
 
   envelopingMistCast(event: CastEvent){
     // Checking to ensure player has cast Enveloping Mists and has the mana reduction buff
-    if (this.selectedCombatant.hasBuff(SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.id)) {
-      this.manaSaved += SPELLS.ENVELOPING_MIST.manaCost * (SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.manaPercRed);
-      this.manaSavedEnm += SPELLS.ENVELOPING_MIST.manaCost * (SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.manaPercRed);
-      this.castsRedEnm += 1;
-      debug && console.log('ENM Reduced');
+    if(this.selectedCombatant.hasBuff(SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.id)) {
```

Comment:
seems like a big pain to do it this way.
I feel like if you order the code a bit it could be clear with less nested if statements. A first one would be to just move the innervate check to the very start of the method.